src/test/java/reposense/authorship/AnnotatorAnalyzerTest.java

```diff
@@ -228,15 +232,15 @@ public void extractAuthorName_invalidAuthorName_returnNull() {
 
         line = "% @@author thisAuthorNameHasMoreThanThirtyNineLetters";
         Assertions.assertEquals(4, AnnotatorAnalyzer.getCommentTypeIndex(line));
-        Assertions.assertFalse(AnnotatorAnalyzer.extractAuthorName(line).isPresent());
+        Assertions.assertTrue(AnnotatorAnalyzer.extractAuthorName(line).isPresent());
 
-        line = "# @@author -invalidUsernameFormat";
+        line = "# @@author -validUsernameFormat";
         Assertions.assertEquals(2, AnnotatorAnalyzer.getCommentTypeIndex(line));
-        Assertions.assertFalse(AnnotatorAnalyzer.extractAuthorName(line).isPresent());
+        Assertions.assertTrue(AnnotatorAnalyzer.extractAuthorName(line).isPresent());
 
         line = "/*@@author fakeAuthor-->";
         Assertions.assertEquals(1, AnnotatorAnalyzer.getCommentTypeIndex(line));
-        Assertions.assertFalse(AnnotatorAnalyzer.extractAuthorName(line).isPresent());
+        Assertions.assertTrue(AnnotatorAnalyzer.extractAuthorName(line).isPresent()); // author name is "fakeAuthor-->"
```

Comment:
Would it make more sense to remove the `invalidAuthorName` test methods altogether? And perhaps add the stranger but now accepted author names like `--authorname--` to the `matchCommentPattern` test cases to ensure that they are parsed correctly.